### PR TITLE
added states

### PR DESCRIPTION
**Pull request by djosborne**

### CODE REVIEW

**Comment by djosborne** [comment]
@nahiluhmot

File: app/cappy/models/state.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module Cappy
+  module Models
+    # This model represents the state of a connected device
+    class State < ActiveRecord::Base
+      self.table_name = 'states'
+      valid_sources = %w(scheduled parent_left manual_override)
```
Comment (nahiluhmot) [inline app/cappy/models/state.rb]: Same here, should be capitalized.

File: spec/app/cappy/models/state_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+describe Cappy::Models::State do
+  describe '#valid?' do
+    let(:device_id) { 'DEADBEEF' }
+    let(:state) { 1 }
+    let(:source) { "scheduled" }
+    let(:created_at) { DateTime.now }
+
+    valid_sources = %w(scheduled parent_left manual_override)
+
+    subject { described_class.new(device_id: device_id,
```
Comment (nahiluhmot) [inline spec/app/cappy/models/state_spec.rb]: Same here, use `do`

**Comment by nahiluhmot** [comment]
@djosborne same two comments. Sorry I'm so pedantic.

**Comment by tlunter** [comment]
The Alert module was going to report a value and a unit.  `State` doesn't look capable of handling that.  Any ideas on that?

**Comment by nahiluhmot** [comment]
@tlunter Good catch on its exclusion. Since each device will only have one kind of unit, perhaps that logic belongs on `Cappy::Models::Device`. It could either be a database field or a hard-coded `Hash` that maps the `device_type` to a `unit`.

@tlunter @djosborne What do you guys think?

**Comment by tlunter** [comment]
I definitely don't want it to be a hard coded hash.  Are we going to try and shoehorn that into the state value? This value should be a float, not an int type.

**Comment by nahiluhmot** [comment]
@tlunter How about decimal to keep precision?

Against the `Hash` in the event that we switch the unit?

**Comment by tlunter** [comment]
Yes, I think the unit should come from the device instead of the backend.

**Comment by nahiluhmot** [comment]
Fair enough.

**Comment by nahiluhmot** [comment]
@djosborne Could you add a `string` field called `unit` to the `devices` table? Non-null, of course.

**Comment by nahiluhmot** [comment]
@djosborne Also, update `status`'s `value` attribute to be a `decimal`.

**Comment by tlunter** [comment]
@nahiluhmot how will we then deal with on off? Using a decimal for that is just as confusing.

**Comment by nahiluhmot** [comment]
@tlunter Same as a float IMO. `0` is off, `1` (or anything else) is on. We can coerce the value to a `bool` in `as_json`.

**Comment by nahiluhmot** [comment]
Or in the devices service.

**Comment by djosborne** [comment]
I'll add the unit field in a new PR

**Comment by djosborne** [comment]
> Also, update ```status```'s ```value``` attribute to be a ```decimal```.

Typo? Do you mean update ```state```'s ```state``` attribute to be decimal?

**Comment by nahiluhmot** [comment]
@djosborne Yeah, sorry about that.

**Comment by djosborne** [comment]
@nahiluhmot  This look good? Like I said, I'm going to add the missing ```unit``` field and add the ```state``` attribute in a new PR. So anything besides that?

File: spec/app/cappy/models/state_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
+require 'spec_helper'
+require 'date'
+
+describe Cappy::Models::State do
+  describe '#valid?' do
+    let(:device_id) { 'DEADBEEF' }
+    let(:state) { 1 }
+    let(:source) { 'scheduled' }
+
+    valid_sources = %w(scheduled parent_left manual_override)
```
Comment (nahiluhmot) [inline spec/app/cappy/models/state_spec.rb]: Remove this, redundant with the `Cappy::Models::State::VALID_SOURCES` constant.

File: spec/app/cappy/models/state_spec.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          context 'but the source is an invalid value' do
+            let(:source) { 'new_source_type' }
+
+            it 'is not valid' do
+              expect(subject).to_not be_valid
+            end
+          end
+
+          context 'and the source is a valid value' do
+            valid_sources.each do |v_s|
```
Comment (nahiluhmot) [inline spec/app/cappy/models/state_spec.rb]: Use the aforementioned constant here.

File: db/migrate/20150125211219_create_states.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class CreateStates < ActiveRecord::Migration
+  def change
+    create_table :states do |t|
+      t.string   :device_id,  null: false
+      t.integer  :state,      null: false
+      t.string   :source,     null: false
+      t.timestamps            null: false
+
+      t.index :device_id
```
Comment (nahiluhmot) [inline db/migrate/20150125211219_create_states.rb]: Add a `unique: true` here.

Comment (nahiluhmot) [inline db/migrate/20150125211219_create_states.rb]: Nevermind, definitely don't do that.

**Comment by nahiluhmot** [comment]
@djosborne Last few comments, then +1

File: spec/app/cappy/models/state_spec.rb (hunk 3 above)
Comment (djosborne) [inline spec/app/cappy/models/state_spec.rb]: How would you import that?Move it into a helpers file?

Comment (djosborne) [inline spec/app/cappy/models/state_spec.rb]: @nahiluhmot

**Pull request merged**